test-js/unit/pentaho/type/simple.serialization.Spec.js

```diff
@@ -158,10 +224,13 @@ define([
 
       describe("values", function() {
         describe("primitive format (omitFormatted: true)", function() {
-          it("should return the primitive value", function() {
+          it("should return the primitive value when it is not a plain object or wrap it otherwise", function() {
             var spec = value.toSpec({omitFormatted: true, includeType: false});
-
-            expect(spec).toBe(simpleSpec.value);
+            if(simpleSpec.value instanceof Object && simpleSpec.value.constructor === Object) {
```

Comment:
🔢 I would go for option 4 (extract the singular behaviour of the plain object) or 5 (maybe with a simple comment somewhere).